File: BackEnd/CasottoIdS/src/main/java/it/unicam/cs/CasottoIdS/models/Evento.java
package it.unicam.cs.CasottoIdS.models;

import org.springframework.data.annotation.Id;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.ArrayList;
import java.util.List;

public class Evento {

    @Id
    public String idEvento;
    private String nome;
    private LocalDate data;
    private LocalTime orarioInizio;
    
    private int numeroMaxPartecipanti;
    private int numeroPartecipanti;

    public Evento(String nome, LocalDate data, LocalTime orarioInizio, int numeroMaxPartecipanti) {
        this.nome = nome;
        this.data = data;
        this.orarioInizio = orarioInizio;
        this.numeroMaxPartecipanti = numeroMaxPartecipanti;
        this.numeroPartecipanti = 0;
    }

   /* public void addPartecipanti(Utente nuovoUtente) {
        this.partecipanti.add(nuovoUtente);
    } */




    public String getIdEvento() {
        return idEvento;
    }

    public void setIdEvento(String idEvento) {
        this.idEvento = idEvento;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public LocalDate getData() {
        return data;
    }

    public void setData(LocalDate data) {
        this.data = data;
    }

    public LocalTime getOrarioInizio() {
        return orarioInizio;
    }

    public void setOrarioInizio(LocalTime orarioInizio) {
        this.orarioInizio = orarioInizio;
    }

    public int getNumeroPartecipanti() {
        return numeroPartecipanti;
    }

    public void setNumeroPartecipanti(int numeroPartecipanti) {
        this.numeroPartecipanti = numeroPartecipanti;
    }

    /* public List<Utente> getPartecipanti() {
        return partecipanti;
    }

    public void setPartecipanti(List<Utente> partecipanti) {
        this.partecipanti = partecipanti;
    } */

    public int getNumeroMaxPartecipanti() {
        return numeroMaxPartecipanti;
    }

    public void setNumeroMaxPartecipanti(int numeroMaxPartecipanti) {
        this.numeroMaxPartecipanti = numeroMaxPartecipanti;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Evento evento = (Evento) o;
        return idEvento.equals(evento.idEvento);
    }


}

File: FrontEnd/Casotto/lib/models/Evento.dart
import 'package:flutter/material.dart';

class Evento {
  late String idEvento;
  late String nome;
  late DateTime data;
  late TimeOfDay orarioInizio;
  late int numeroMaxPartecipanti;
  late int numeroPartecipanti;

  Evento(String idEvento, String nome, DateTime data, TimeOfDay orarioInizio,
      int numeroMaxPartecipanti, int numeroPartecipanti) {
    this.idEvento = idEvento;
    this.nome = nome;
    this.data = data;
    this.orarioInizio = orarioInizio;
    this.numeroMaxPartecipanti = numeroMaxPartecipanti;
    this.numeroPartecipanti = numeroPartecipanti;
  }

  getIdEvento() {
    return idEvento;
  }

  void setIdEvento(String idEvento) {
    this.idEvento = idEvento;
  }

  String getNome() {
    return nome;
  }

  void setNome(String nome) {
    this.nome = nome;
  }

  DateTime getData() {
    return data;
  }

  String getDataString() {
    return "${this.data.day}-${this.data.month}-${this.data.year}";
  }

  void setData(DateTime data) {
    this.data = data;
  }

  TimeOfDay getOrarioInizio() {
    return orarioInizio;
  }

  String getOrarioInizioString() {
    return "${this.orarioInizio.hour}:${this.orarioInizio.minute}";
  }

  void setOrarioInizio(TimeOfDay orarioInizio) {
    this.orarioInizio = orarioInizio;
  }

  int getNumeroPartecipanti() {
    return numeroPartecipanti;
  }

  void setNumeroPartecipanti(int numeroPartecipanti) {
    this.numeroPartecipanti = numeroPartecipanti;
  }

  int getNumeroMaxPartecipanti() {
    return numeroMaxPartecipanti;
  }

  void setNumeroMaxPartecipanti(int numeroMaxPartecipanti) {
    this.numeroMaxPartecipanti = numeroMaxPartecipanti;
  }

  @override
  String toString() {
    return "Evento{" +
        "idEvento=" +
        idEvento +
        ", nome=" +
        nome +
        ", Data=" +
        data.toString() +
        "Orario inizio: " +
        orarioInizio.toString() +
        "numero max partecipanti" +
        numeroMaxPartecipanti.toString() +
        "numero partecipanti: " +
        numeroPartecipanti.toString() +
        " }";
  }
}
